agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,195 @@
+
+echo "$person"
+echo "$helloDurationMilliseconds"
+echo "$helloCount"
+
+# Get the number of retries allowed in case of a failing test
+retries=$(test-agent-cli retry-count)
+echo "no of retries are $retries"
+
+# Get the secret for a key
+secret=$(test-agent-cli get-secret key1)
+echo "Value for secret is $secret"
+
+# Perform the test with this bash script.
+
+test () {
+for ((i = 1; i <= helloCount; i++))
+do
+echo "Hello ${i} to ${person}"
+sleep "${helloDurationMilliseconds}"
+done
+}
+
+# Retry the test in case of failure
+if ! test && [ "$retries" -gt 0 ]; then
+for ((i = 1; i <= "$retries"; i ++))
+do
+echo "Test failed again, retrying ${i} of $retries"
+
+# This will save the result of current iteration in a directory on kubernetes cluster,
+# that will contain the info how many test cases failed, passed, skipped.
+test-agent-cli send-result -o fail -p 0 -f 1 -s 0
+
+if test; then
+    break
+fi
+done
+fi
+
+# Save the test result after consuming all the retries,  that will contain
+# the info how many test cases failed, passed, skipped
+test-agent-cli send-result -o pass -p 1 -f 0 -s 0
+
+# Set the Task_state as completed. This command will also create the tar file of the results
+# This will change the resource status in testsys to passed/failed, but Pod state will remain running if keep_running has been set as true
+# To change the pod state to complete set keep_running as false(testsys set test_name --keep-running false)
+test-agent-cli terminate --results-dir results_directory
+
+# Setting the task_status and resource status as error if some error encounters
+# test-agent-cli send-error error_encountered
+```
+
+2. Create a docker file, that will be used to create image that will be running on the Test Pod. Following is an example of the Docker file that provides all the required resources to run the Bash script along with the Command Line Interface in a single pod. In this docker file make sure to mark entry point as the Test Bash script created in Step 1.
+
+```shell
+# syntax=docker/dockerfile:1.1.3-experimental
+
+ARG BUILDER_IMAGE
+FROM ${BUILDER_IMAGE} as build
+
+ARG ARCH
+USER root
+
+# We need these environment variables set for building the `openssl-sys` crate
+
+ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
+ENV PKG_CONFIG_ALLOW_CROSS=1
+ENV CARGO_HOME=/src/.cargo
+ENV OPENSSL_STATIC=true
+ENV CARGO_HOME=/src/.cargo
+ADD ./ /src/
+WORKDIR /src/agent/test-agent-cli
+RUN --mount=type=cache,mode=0777,target=/src/target 
+cargo install --locked --offline --target ${ARCH}-bottlerocket-linux-musl --path .  --root ./
+
+FROM public.ecr.aws/amazonlinux/amazonlinux:2
+RUN yum update -y 
+&& yum install -y jq 
+&& yum clean all
+COPY --from=build /src/agent/test-agent-cli/bin/test-agent-cli /usr/local/bin/
+COPY --from=build /src/agent/test-agent-cli/examples/example_test_agent_cli/example-test.sh ./
+
+ENTRYPOINT ["/bin/bash", "./example-test.sh"]
+```
```

Comment:
Can you explain what is boilerplate and what is configurable.

Reply:
I have added that whatever they need to run Bash script, they can add that in docker file. Should I mention that The way to copy test agent cli, example test and other part of the file should remain same?